webapp/src/ts/modules/tasks/tasks-group.component.ts

```diff
@@ -0,0 +1,260 @@
+      if (!this.contactModel) {
+        this.navigationCancel();
+        return;
+      }
+
+      const idsForTasks = this.tasksForContactService.getIdsForTasks(this.contactModel);
+      if (!idsForTasks || !idsForTasks.length) {
+        this.navigationCancel();
+        return;
+      }
+
+      const filteredTasks = this.cherryPickTasks(idsForTasks);
+      this.recordTaskCountTelemetry(this.contactModel);
+      if (!filteredTasks || !filteredTasks.length) {
+        this.navigationCancel(true);
+        return;
+      }
+
+      this.setNavigation(true);
+      this.tasks = filteredTasks;
+      this.globalActions.setLoadingContent(false);
+    } catch (err) {
+      console.error('Error loading tasks group', err);
+      this.contentError = true;
+      this.globalActions.setLoadingContent(false);
+      this.setNavigation(false);
+    }
+  }
+
+  private recordTaskCountTelemetry(contactModel) {
+    if (this.telemetryRecorded) {
+      return;
+    }
+
+    this.ngZone.runOutsideAngular(async () => {
+      this.telemetryRecorded = true;
+      const taskCounts = await this.tasksForContactService.getTasksBreakdown(contactModel);
+      let allTasksCount = 0;
+      Object.keys(taskCounts).forEach(state => {
+        // when cherry picking, we might exclude the last completed task, if still present in the list
+        const count = state === 'Ready' ? this.tasks.length : taskCounts[state];
+        allTasksCount += count;
+      });
+
+      this.telemetryService.record('tasks:group:all-tasks', allTasksCount);
+      this.telemetryService.record('tasks:group:cancelled', taskCounts.Cancelled);
+      this.telemetryService.record('tasks:group:ready', this.tasks.length);
+      const breakdownByTitle:any = { };
+      this.tasks.forEach(({ title }) => {
+        if (Object.hasOwnProperty.call(breakdownByTitle, title)) {
+          breakdownByTitle[title]++;
+        } else {
+          breakdownByTitle[title] = 1;
+        }
+      });
+      Object.keys(breakdownByTitle).forEach(title => {
+        this.telemetryService.record(`tasks:group:ready:${title}`, breakdownByTitle[title]);
+      });
+    });
+  }
+
+  private isGroupTask(emissionId) {
+    return Array.isArray(this.tasks) && this.tasks.find((task:any) => task?._id === emissionId);
+  }
+
+  canDeactivate(nextUrl) {
+    if (!this.tasks || !this.tasks.length || !this.cancelCallback || !this.preventNavigation) {
+      return true;
+    }
+
+    const emissionId = this.getEmissionIdFromUrl(nextUrl);
+    if (emissionId && this.isGroupTask(emissionId)) {
+      return true;
+    }
+
+    this.globalActions.navigationCancel(nextUrl);
+    return false;
+  }
+
+  private getEmissionIdFromUrl(url) {
+    const tree = this.router.parseUrl(url);
+    // https://angular.io/api/router/UrlTree#example
+    const segments = tree.root.children[PRIMARY_OUTLET].segments;
+    if (segments && segments.length === 2 && segments[0]?.path === 'tasks') {
+      return segments[1]?.path;
```

Comment:
This works nicely! It required a few more changes, however, to make sure that the routes we try to access have the extras. 
It also makes the unit tests more fragile, because now we have to mock the router's current navigation instead of just parsing a url, so we depend on the e2e test to verify this interaction.

Reply:
Ok, I went with adding a `state` with a `tab` and `id` property, and added it to the task list and content row list item. 
I didn't add the properties to the other lists, but we could add them as we need them.